### ISSUE TITLE
async d2h and h2d copy issue in HIP

### ISSUE THREAD

**Issue by luraess** [eth-cscs/ImplicitGlobalGrid.jl#63]
In IGG v0.13, `write_d2h_async!` and `write_h2d_async!` for ROCArrays (AMDGPU) need a fix.

Currently fixing the error when `pos!=(0,0,0)` results in an offset by one in the copied results (see https://github.com/ROCm-Developer-Tools/HIP/issues/3289).

**Comment by omlins** [comment]
@luraess : do you have a fix for this meanwhile?

**Comment by luraess** [comment]
No, I pinged the ROCm issue https://github.com/ROCm-Developer-Tools/HIP/issues/3289 but meanwhile we can't have the async copy.

**Comment by omlins** [comment]
@luraess : what is the status of this?

**Comment by luraess** [comment]
> @luraess : what is the status of this?

Should be fixed upstream by now https://github.com/ROCm/HIP/issues/3289#issuecomment-1920700024

**Comment by luraess** [comment]
Given that it's now fixed upstream, we could roll back the async copy 3D features commented out in here https://github.com/eth-cscs/ImplicitGlobalGrid.jl/blob/1558da4c53c7e34145d15fa2b08f7815d92c33f2/src/AMDGPUExt/update_halo.jl#L153 and https://github.com/eth-cscs/ImplicitGlobalGrid.jl/blob/1558da4c53c7e34145d15fa2b08f7815d92c33f2/src/AMDGPUExt/update_halo.jl#L184

**Comment by omlins** [comment]
> Given that it's now fixed upstream, we could roll back the async copy 3D features commented out in here 

@luraess could you please do that to finish this?
